packages/server-wallet/src/wallet/store.ts

```diff
@@ -391,14 +400,37 @@ export class Store {
         return this.ensureOpenChannelObjective(objective, tx);
       case 'CloseChannel':
         return this.ensureCloseChannelObjective(objective, tx);
-
       case 'SubmitChallenge':
         return this.ensureSubmitChallengeObjective(objective, tx);
+      case 'DefundChannel':
+        return this.ensureDefundChannelObjective(objective, tx);
       default:
         throw new StoreError(StoreError.reasons.unimplementedObjective);
     }
   }
 
+  private async ensureDefundChannelObjective(
+    objective: DefundChannel,
+    tx: Transaction
+  ): Promise<DBObjective> {
+    const {data} = objective;
+    const {targetChannelId} = data;
+    // fetch the channel to make sure the channel exists
+    const channel = await this.getChannelState(targetChannelId, tx);
+    if (!channel) {
+      throw new StoreError(StoreError.reasons.channelMissing, {channelId: targetChannelId});
+    }
+
+    try {
+      await ObjectiveModel.insert({...objective, status: 'pending'}, tx);
+    } catch (e) {
+      // If the objective exists our job is done
+      if (e.name !== 'UniqueViolationError') throw e;
+    }
```

Comment:
I think the `ObjectiveModel` should handle this logic? Right now, the internal model implementation has leaked into the store.

Postgres allows `ON CONFLICT`, so the `ObjectiveModel` can do, for instance,
```
INSERT foo
ON CONFLICT (name of uniqueness constraint) DO NOTHING
```

Reply:
I agree with this. Mea culpa: I just copied this from the existing `ensure` methods.

Reply:
Happy to leave this as a future refactoring work, if the pattern exists elsewhere.